bench/helpers.go

```diff
@@ -0,0 +1,161 @@
+// offers
+// https://github.com/topfreegames/offers
+//
+// Licensed under the MIT license:
+// http://www.opensource.org/licenses/mit-license
+// Copyright © 2016 Top Free Games <backend@tfgco.com>
+
+package bench
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/satori/go.uuid"
+	"github.com/topfreegames/offers/models"
+	"gopkg.in/mgutz/dat.v2/dat"
+	runner "gopkg.in/mgutz/dat.v2/sqlx-runner"
+)
+
+func getRoute(url string) string {
+	return fmt.Sprintf("http://localhost:8888%s", url)
+}
+
+func createGames(db *runner.Connection, numberOfGames int) ([]*models.Game, error) {
+	var games []*models.Game
+
+	for i := 0; i < numberOfGames; i++ {
+		now := time.Now()
+		game := &models.Game{
+			Name: fmt.Sprintf("game-%d", i),
+			ID:   uuid.NewV4().String(),
+		}
+		if err := models.UpsertGame(*db, game, now, nil); err != nil {
+			return nil, err
+		}
+		games = append(games, game)
+	}
+
+	return games, nil
+}
+
+func createGame(db *runner.Connection) (*models.Game, error) {
+	game := &models.Game{
+		Name: "game-name",
+		ID:   uuid.NewV4().String(),
+	}
+	err := models.UpsertGame(*db, game, time.Now(), nil)
+
+	return game, err
+}
+
+func createOffers(db *runner.Connection, game *models.Game, enabled bool, numberOfOffers int) ([]*models.Offer, error) {
+	var err error
+	offers := getOffers(game, enabled, numberOfOffers)
+
+	for _, offer := range offers {
+		offer, err = models.InsertOffer(*db, offer, nil)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return offers, nil
+}
+
+func getOffers(game *models.Game, enabled bool, numberOfOffers int) []*models.Offer {
+	var offers []*models.Offer
+
+	for i := 0; i < numberOfOffers; i++ {
+		to := time.Now().Unix() + 5*60
+		trigger := fmt.Sprintf("{\"from\": 1486678000, \"to\": %d}", to)
+		offer := &models.Offer{
+			GameID:    game.ID,
+			Name:      fmt.Sprintf("offer-%d", i),
+			Period:    dat.JSON([]byte(`{"every": "1h"}`)),
+			Frequency: dat.JSON([]byte(`{"every": "1h"}`)),
+			Trigger:   dat.JSON([]byte(trigger)),
+			Placement: "popup",
+			ProductID: "tfg.com.sample",
+			Contents:  dat.JSON([]byte(`{"x": 1}`)),
+			Enabled:   enabled,
+		}
+		offers = append(offers, offer)
+	}
+
+	return offers
+}
+
+func createOffer(db *runner.Connection, game *models.Game) (*models.Offer, error) {
+	var err error
+	to := time.Now().Unix() + 5*60
```

Comment:
Same here. You could use the method getOffers with numberOfOffers=1 and avoid all this repeated code.